samples/siwx91x_ota/src/main.c

```diff
@@ -0,0 +1,756 @@
+	} else {
+		printf("Disconnection reason: %d\n", status->disconn_reason);
+		ctx->disonnected = true;
+	}
+}
+
+static void ota_wifi_mgmt_event_handler(struct net_mgmt_event_callback *cb, uint64_t mgmt_event,
+					struct net_if *iface)
+{
+
+	switch (mgmt_event) {
+	case NET_EVENT_WIFI_SCAN_RESULT:
+		ota_handle_wifi_scan_result(cb);
+		break;
+	case NET_EVENT_WIFI_SCAN_DONE:
+		ota_handle_wifi_scan_done(cb);
+		break;
+	case NET_EVENT_WIFI_CONNECT_RESULT:
+		ota_handle_wifi_connect_result(cb);
+		break;
+	case NET_EVENT_WIFI_DISCONNECT_RESULT:
+		ota_handle_wifi_disconnect_result(cb);
+		break;
+	default:
+		break;
+	}
+}
+
+static int ota_wifi_start_scan(struct app_ctx *ctx)
+{
+	struct net_if *iface = net_if_get_first_wifi();
+	struct wifi_scan_params params = {};
+	int ret;
+
+	printf("%-4s | %-32s %-5s | %-13s | %-4s | %-15s | %-17s | %-8s\n", "Num", "SSID", "(len)",
+	       "Chan (Band)", "RSSI", "Security", "BSSID", "MFP");
+	ret = net_mgmt(NET_REQUEST_WIFI_SCAN, iface, &params, sizeof(params));
+	if (ret) {
+		return ret;
+	}
+	return 0;
+}
+
+/*
+ * @brief Connect to WiFi network
+ *
+ * @param ssid Network SSID
+ * @param pwd Network password
+ * @param ctx App context pointer
+ * @return int 0 on success, negative error code on failure
+ */
+static int ota_wifi_connect(const char *ssid, const char *pwd, struct app_ctx *ctx)
+{
+	struct wifi_connect_req_params cnx_params = {
+		.channel = WIFI_CHANNEL_ANY,
+		.band = 0,
+		.security = ctx->security_type,
+		.psk_length = strlen(pwd),
+		.psk = (uint8_t *)pwd,
+		.ssid_length = strlen(ssid),
+		.ssid = (uint8_t *)ssid
+	};
+	struct net_if *iface = net_if_get_first_wifi();
+	int ret;
+
+	if (!ssid || !pwd) {
+		printf("Invalid connection parameters\n");
+		return -EINVAL;
+	}
+
+	printf("Connecting to network: %s\n", ssid);
+	ret = net_mgmt(NET_REQUEST_WIFI_CONNECT, iface, &cnx_params,
+		       sizeof(struct wifi_connect_req_params));
+	if (ret) {
+		printf("Connection request failed with error: %d\n", ret);
+		return ret;
+	}
+	return 0;
+}
+
+/*
+ * @brief Wait for IP configuration to complete
+ * @param ctx Pointer to application context structure
+ * @return int 0 on success, negative error code on failure
+ */
+static int ota_wait_for_ip_config(struct app_ctx *ctx)
+{
+	int ret;
+
+	if (IS_ENABLED(CONFIG_NET_IPV4)) {
+		ret = k_event_wait_safe(&ctx->events, NET_IPV4_EVENT_READY, false, K_SECONDS(30));
+	} else {
+		ret = k_event_wait_safe(&ctx->events, NET_IPV6_EVENT_READY, false, K_SECONDS(30));
+	}
+	if (!ret && IS_ENABLED(CONFIG_NET_IPV6)) {
+		ret = k_event_wait_safe(&ctx->events, NET_IPV6_EVENT_READY, false, K_NO_WAIT);
+	}
+	return (ret > 0) ? 0 : -ETIMEDOUT;
+}
+
+/*
+ * @brief Connect to OTA server
+ *
+ * @param ctx Pointer to application context structure
+ * @return int 0 on success, negative error code on failure
+ */
+static int ota_connect_to_server(struct app_ctx *ctx)
+{
+	struct http_parse_data http_parse_data_info = ctx->http_parse_data_info;
```

Comment:
`data` and `info` are redundant. BTW, why don't you get a pointer rather than duplicating this structure?

     struct url_data *url_data = &ctx->url_data;